api/users/views.py

```diff
@@ -300,6 +300,7 @@ class UserNodes(JSONAPIBaseView, generics.ListAPIView, UserMixin, ODMFilterMixin
     # overrides ODMFilterMixin
     def get_default_odm_query(self):
         user = self.get_user()
+        user.update_affiliated_institutions_by_email_domain()
```

Comment:
This probably isn't the place to so this; a `GET` request should not result in writes to the database.

How about putting this in the view for the Dashboard?